File: src/src/ViazyNetCore.Model/Shop/ProductOuter.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ViazyNetCore.Model
{
    [Table(Name = "ShopMall.ProductOuter")]
    public class ProductOuter : EntityBase<string>
    {

        /// <summary>
        /// 设置或获取一个值，表示类型名称。
        /// </summary>
        public string OuterName { get; set; }

        /// <summary>
        /// 设置或获取一个值，表示类型类别。
        /// </summary>
        public string OuterType { get; set; }

        public ComStatus Status { get; set; }

        /// <summary>
        /// 设置或获取一个值，表示类型描述。
        /// </summary>
        public string Description { get; set; }

        /// <summary>
        /// 设置或获取一个值，表示类型启用时间。
        /// </summary>
        public DateTime? BeginTime { get; set; }

        /// <summary>
        /// 设置或获取一个值，表示类型结束时间。
        /// </summary>
        public DateTime? EndTime { get; set; }

        public DateTime CreateTime { get; set; }
    }
}

File: fontend/ele-admin-ui/src/api/shopmall/productOuter.ts

  import { ComStatus,Pagination } from '../model'

  
  /**
  * ProductOuterPageData
  */
  export interface ProductOuterPageData {
    
    rows: Array<ProductOuter>
    
    total: number
  }
  
  /**
  * ProductOuter
  */
  export interface ProductOuter {
    
    outerName: string
    
    outerType: string
    
    status: ComStatus
    
    description: string
    
    beginTime: string
    
    endTime: string
    
    createTime?: string
    
    id?: string
  }
  

  import { http } from '@/utils/http';
  
  export class ProductOuterApi {
    /**
     * 无
     */
    public apiProductOuterFindAll(param1?: Pagination): Promise<ProductOuterPageData> {
      return http.request({
        url: '/api/ProductOuter/FindAll',
        method: 'post',
        data: param1
      })
    }
    public apiProductOuterModifyStatus(outerId?: string, status?: ComStatus): Promise<boolean> {
        return http.request({
          url: '/api/ProductOuter/ModifyStatus',
          method: 'post',
          params: { outerId, status }
        })
      }
      public apiProductOuterMangerProductOuter(param1?: ProductOuter): Promise<boolean> {
        return http.request({
          url: '/api/ProductOuter/MangerProductOuter',
          method: 'post',
          data: param1
        })
      }
  }
  export default new ProductOuterApi()
  